capsule_network.py

```diff
@@ -43,16 +43,18 @@ def __init__(self,
                                     unit_size=primary_unit_size,
                                     use_routing=False)
 
-        self.digits = CapsuleLayer(in_units=num_primary_units,
-                                   in_channels=primary_unit_size,
+        self.digits = CapsuleLayer(in_units=primary_unit_size,
+                                   in_channels=num_primary_units,
                                    num_units=num_output_units,
                                    unit_size=output_unit_size,
                                    use_routing=True)
 
         reconstruction_size = image_width * image_height * image_channels
-        self.reconstruct0 = nn.Linear(num_output_units*output_unit_size, (reconstruction_size * 2) / 3)
-        self.reconstruct1 = nn.Linear((reconstruction_size * 2) / 3, (reconstruction_size * 3) / 2)
-        self.reconstruct2 = nn.Linear((reconstruction_size * 3) / 2, reconstruction_size)
+        # self.reconstruct0 = nn.Linear(num_output_units*output_unit_size, (reconstruction_size * 2) / 3)
```

Comment:
you should pull my latest changes.  i fixed a problem in the reconstruction loss.

anyway, are you sure this is correct?  the paper (and other implementations i've seen) seem to say that the reconstruction input is *all* capsules, but with the inactive capsules masked out.

Reply:
yes, you're right.  Now I can get 98.9% in first epoch after fix few bug.  But I still wonder why we softmax the wrong dim and squash the wrong dim in routing part can still get such a great result. 
And the most weird thing is that if I make the dim correct this model  would crash sometimes, and get bad accuracy... Hope you can fix it. Thx.